controllers/utils/handlers/enqueueformatchingupdate.go

```diff
@@ -0,0 +1,117 @@
+		nn := parseNamespacedName(releasePlan.Status.ReleasePlanAdmission.Name)
+		if nn.Name != "" {
+			rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+		}
+	} else if releasePlanAdmission, ok := updateEvent.ObjectOld.(*v1alpha1.ReleasePlanAdmission); ok {
+		for _, releasePlan := range releasePlanAdmission.Status.ReleasePlans {
+			nn := parseNamespacedName(releasePlan.Name)
+			if nn.Name != "" {
+				rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+			}
+		}
+	}
+
+	if releasePlan, ok := updateEvent.ObjectNew.(*v1alpha1.ReleasePlan); ok {
+		nn := parseNamespacedName(releasePlan.Status.ReleasePlanAdmission.Name)
+		if nn.Name != "" {
+			rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+		}
+	} else if releasePlanAdmission, ok := updateEvent.ObjectNew.(*v1alpha1.ReleasePlanAdmission); ok {
+		for _, releasePlan := range releasePlanAdmission.Status.ReleasePlans {
+			nn := parseNamespacedName(releasePlan.Name)
+			if nn.Name != "" {
+				rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+			}
+		}
+	}
+}
+
+// Delete implements EventHandler
+func (e *EnqueueForMatchingUpdate) Delete(_ context.Context, deleteEvent event.DeleteEvent, rateLimitingInterface workqueue.RateLimitingInterface) {
+	if releasePlan, ok := deleteEvent.Object.(*v1alpha1.ReleasePlan); ok {
+		nn := parseNamespacedName(releasePlan.Status.ReleasePlanAdmission.Name)
+		if nn.Name != "" {
+			rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+		}
+	} else if releasePlanAdmission, ok := deleteEvent.Object.(*v1alpha1.ReleasePlanAdmission); ok {
+		for _, releasePlan := range releasePlanAdmission.Status.ReleasePlans {
+			nn := parseNamespacedName(releasePlan.Name)
+			if nn.Name != "" {
+				rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+			}
+		}
+	}
+}
+
+// Generic implements EventHandler
+func (e *EnqueueForMatchingUpdate) Generic(_ context.Context, genericEvent event.GenericEvent, rateLimitingInterface workqueue.RateLimitingInterface) {
+	if releasePlan, ok := genericEvent.Object.(*v1alpha1.ReleasePlan); ok {
+		nn := parseNamespacedName(releasePlan.Status.ReleasePlanAdmission.Name)
+		if nn.Name != "" {
+			rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+		}
+	} else if releasePlanAdmission, ok := genericEvent.Object.(*v1alpha1.ReleasePlanAdmission); ok {
+		for _, releasePlan := range releasePlanAdmission.Status.ReleasePlans {
+			nn := parseNamespacedName(releasePlan.Name)
+			if nn.Name != "" {
+				rateLimitingInterface.Add(reconcile.Request{NamespacedName: nn})
+			}
+		}
+	}
+}
+
+// parseNamespacedName parses the provided string to extract the namespace and name into a
+// types.NamespacedName.
+func parseNamespacedName(namespacedNameString string) types.NamespacedName {
+	values := strings.SplitN(namespacedNameString, "/", 2)
+	if len(values) < 2 {
+		return types.NamespacedName{}
+	}
+	return types.NamespacedName{Namespace: values[0], Name: values[1]}
+
```

Comment:
For once you add an empty line but it's in the wrong place :stuck_out_tongue_winking_eye: 

Please, surround the if with empty lines.

Reply:
fixed